scripts/release.py

```diff
@@ -228,31 +249,93 @@ def ensure_clean_git() -> None:
         sys.exit(1)
 
 
-def parse_next_version(version: Text) -> Text:
+def validate_version(version: Text) -> bool:
+    """
+    Ensure that the version follows semver
+    and that the prerelease follows the format `a1`, `rc2`, etc...
+    """
+    if isinstance(pep440_version.parse(version), pep440_version.LegacyVersion):
+        return False
+
+    version_object = Version.coerce(version)
+    return not version_object.prerelease or is_prerelease_version(version_object)
+
+
+def is_prerelease_version(version: Version) -> bool:
+    """
+    Validate that the prerelease part in a version follows
+    the pattern specified in `PRERELEASE_VERSION_PATTERN`.
+    """
+    return (
+        len(version.prerelease) == 1
+        and PRERELEASE_VERSION_PATTERN.match(version.prerelease[0]) is not None
+    )
+
+
+def is_alpha_version(version: Version) -> bool:
+    """
+    Validate that the alpha part in a version follows
+    the pattern specified in `PRERELEASE_VERSION_PATTERN`
+    and is an alpha (as opposed to a release candidate).
+    """
+    if len(version.prerelease) != 1:
+        return False
+
+    version_match = PRERELEASE_VERSION_PATTERN.match(version.prerelease[0])
+    if version_match is None:
+        return False
+
+    return version_match.group(1) == "a"
+
+
+def next_prerelease(version: Version, flavor: Text) -> Version:
+    """Bump the current version to the next prerelease."""
+    prerelease_number = 0
+    if version.prerelease:
+        prerelease_number = int(
+            PRERELEASE_VERSION_PATTERN.match(version.prerelease[0]).group(2)
+        )
+
+    return Version(
+        major=version.major,
+        minor=version.minor,
```

Comment:
thinking about this, the increase of the prerelease is a bit more complicated:
- current version `1.2.3a2` --> next prerelease is `1.2.3a3` (this should already work)
- current version `1.2.3` (base branch `master`) --> next wanted prerelease is probably `1.3.0a1`
- current version `1.2.3` (base branch is release branch, `1.2.x`) --> next wanted prerelease is probably `1.2.4a1`

same for `rc`, if current version is already a prerelease version, just change the prerelease part, if not the version part needs to be changed as well.

Reply:
or alternatively to the branch logic, we could add another question if the user selects `patch` / `rc`: 
`What kind of prerelease do you want?` patch | minor | major (not sure how to phrase though, would also only be needed if the current version is not prerelease already)

Reply:
🤔 what about:
- if you select alpha/rc and the current version is `1.2.3`, ask if we want to bump to `1.2.3a1` `1.2.4a1` or `1.3.0a1` (and the equivalent for rc?). We could even include `2.0.0a1`.
- if you select alpha/rc and the current version is already an alpha/rc, just increate the number

I would avoid a complex flow here and defer to user choice.

Reply:
yes that sounds perfect :+1: luckily there is `questionary.select` 😂

Reply:
i would not include `1.2.3a1` in your above example, it doesn't make sense to release a "prerelease" version after the "final" release of that version (`1.2.3` in this case). I think the three choices should be `1.2.4a1`, `1.3.0a1` and `2.0.0a1`

Reply:
Would look like that for alpha:
<img width="1065" alt="Screenshot 2020-05-15 at 16 34 56" src="https://user-images.githubusercontent.com/1931615/82062234-19490f00-96ca-11ea-99c1-217b65a6871c.png">
Would look like that for rc:
<img width="1048" alt="Screenshot 2020-05-15 at 16 35 09" src="https://user-images.githubusercontent.com/1931615/82062236-1bab6900-96ca-11ea-87c0-f5f9deaf3608.png">

btw I your `questionary` lib a lot!

Reply:
looks good :+1: (as said, would remove the `1.10.0a1` - esp. since it might already exist)

Reply:
and can you make the one that bumps the minor the default? it is going to be the most common case anyways

Reply:
👍

Reply:
(GH comments didn't refresh...)

Reply:
<img width="387" alt="Screenshot 2020-05-15 at 17 00 06" src="https://user-images.githubusercontent.com/1931615/82064844-96c24e80-96cd-11ea-924d-21df080b8cc4.png">

The `default` parameter seems broken, reordering the choices so that the first choice is the minor bump ✅